scripts/ai/BaleToCollect.lua

```diff
@@ -136,6 +136,59 @@ function BaleToCollect:getSafeDistance()
 	return math.sqrt(length * length + self.bale.width * self.bale.width) / 2 + 0.2
 end
 
-function BaleToCollect.getAllBales()
+---@class BaleToCollectManager
+BaleToCollectManager = CpObject()
+BaleToCollectManager.lockTimeOutMs = 500 -- 500 ms
+
+function BaleToCollectManager:init()
+	self.temporaryLockedBales = {}
+	self.lockedBales = {}
+end
+
+function BaleToCollectManager:update(dt)
+	for bale, time in pairs(self.temporaryLockedBales) do 
+		if time < (g_time + self.lockTimeOutMs) then 
+			self.temporaryLockedBales[bale] = nil

```

Comment:
I would have done this with a CpTemporaryObject so there is no need for a cyclic update.